postfix_mta_sts_resolver/proactive_fetcher.py

```diff
@@ -0,0 +1,113 @@
+import asyncio
+import logging
+import time
+
+from postfix_mta_sts_resolver import constants
+from postfix_mta_sts_resolver.base_cache import CacheEntry
+from postfix_mta_sts_resolver.resolver import STSResolver, STSFetchResult
+
+
+class STSProactiveFetcher:
+    def __init__(self, cfg, loop, cache):
+        self._shutdown_timeout = cfg['shutdown_timeout']
+        self._proactive_fetch_interval = cfg['proactive_fetch_interval']
+        self._logger = logging.getLogger("PF")
+        self._loop = loop
+        self._cache = cache
+        self._periodic_fetch_task = None
+        self._resolver = STSResolver(loop=loop,
+                                     timeout=cfg["default_zone"]["timeout"])
+
+    async def process_domain(self, domain_queue):
+        while True:  # Run until cancelled
+            cache_item = await domain_queue.get()
+            try:
+                domain, cached = cache_item
+                status, policy = await self._resolver.resolve(domain)
```

Comment:
We should pass known policy version id from cached entry to `resolve()` so resolver will be able to skip actual policy HTTPS fetch if result is not changed.